src/build.d

```diff
@@ -1796,12 +1805,35 @@ Params:
     spec = a format specifier
     args = the data to format to the log
 */
-auto log(T...)(string spec, T args)
+void log(T...)(string spec, T args)
 {
     if (verbose)
         writefln(spec, args);
 }
 
+/**
+Logs multiple lines of output.
+All lines are prefixed with the provided section name.
+
+Params:
+    section = section name to use a line prefix
+    r = output to log
+*/
+void logOutput(R)(string section, R r)
+{
+    import std.ascii : newline;
+    section = section.empty ? "unknown" : section;
+    auto sectionName = text("[", section, "] ");
+    if (r.empty)
+        return;
+    // write as one message to avoid interleaved messages
+    r.stripRight
+     .splitter(newline)
+     .map!(l => chain(sectionName, l))
+     .joiner(newline)
+     .writeln;
```

Comment:
Couldn't this use `lockingTextWriter` instead of joining all lines?

Reply:
The idea was to write the entire output as one block, s.t. there are no messages from other steps in-between.

Reply:
Right, didn't think about other messages written to `stderr`